packages/plugin-github/sources/githubUtils.ts

```diff
@@ -1,27 +1,55 @@
+
 const githubPatterns = [
   /^github:([^\/#]+)\/([^\/#]+)(?:#(.+))?$/,
-  /^git@github.com:([^\/#]+)\/([^\/#]+)(?:#(.+))?$/,
+  /^[git@github.com:|git:\/\/github.com]+([^\/#]+)\/([^\/#]+)\.git(?:#(.+))?$/,
   /^git\+https:\/\/github.com\/([^\/#]+)\/([^\/#]+)\.git?(?:#(.+))?$/,
   /^https:\/\/github.com\/([^\/#]+)\/([^\/#]+)\/tarball\/([^\/#]+)(?:#|$)/,
+  /^[http|https]+:\/\/.*github.com\/([^\/#]+)\/([^\/#]+)\.git?(?:#(.+))?$/,
 ];
 
-export function isGithubUrl(string: string) {
-  return githubPatterns.some(pattern => !!string.match(pattern));
+/**
+  Determines whether a given url is a valid github git url via regex
+*/
+export function isGithubUrl(url: string): boolean {
+  return url ? githubPatterns.some(pattern => !!url.match(pattern)) : false;
+}
+
+const getBranch = (parsedBranch: string): string | undefined => {
+  return parsedBranch ? parsedBranch : undefined;
+}
+
+interface ParsedGithubUrl {
+  username: string;
+  reponame: string;
+  branch?: string;
 }
 
-export function parseGithubUrl(string: string) {
-  let match;
+/**
+ * Takes a valid github repository url and parses it, returning
+ * an object of type `ParsedGithubUrl`
+ */
+export function parseGithubUrl(urlStr: string): ParsedGithubUrl {
+  let match: Array<string> | null = null;
 
   for (const pattern of githubPatterns) {
-    match = string.match(pattern);
+    match = urlStr.match(pattern);
     if (match) {
       break;
     }
   }
 
   if (!match)
-    throw new Error(`Input cannot be parsed as a valid Github URL ('${string}')`);
+    throw new Error(invalidGithubUrlMessage(urlStr));
 
   const [, username, reponame, branch] = match;
-  return {username, reponame, branch: branch ? branch : undefined};
+  return {username, reponame, branch: getBranch(branch)};
+}
+
+export function invalidGithubUrlMessage(url: string): string {
```

Comment:
I don't think we need a detailed error message - this function will only be called if `isGithubUrl` previously returned true. It's more an assertion that a validation 😃

Reply:
:heavy_check_mark: